comms/common/algorithms/all_gather/all_gather_dda.cuh

```diff
@@ -1,8 +1,19 @@
 // (c) Meta Platforms, Inc. and affiliates. Confidential and proprietary.
 
+#if defined(__HIP_PLATFORM_AMD__) || defined(__HIP_PLATFORM_HCC__)
+#include <hip/hip_runtime.h>
+#include <hip/hip_bfloat16.h>
+#include <hip/hip_fp16.h>
+#include <hip/amd_detail/amd_hip_bf16.h>
+using bf16 = __hip_bfloat16;
+using bf162 = __hip_bfloat162;
+#else
 #include <cuda.h>
 #include <cuda_bf16.h>
 #include <cuda_fp16.h>
+using bf16 = __nv_bfloat16;
```

Comment:
Why do we redefine these here if we already have them in comms/common/algorithms/CollCommon.cuh which gets included below?